### ISSUE TITLE
Error during webpack build

### ISSUE THREAD

**Issue by theladyjaye** [DeloitteDigitalAPAC/react-habitat-redux#1]
#1 thanks for this lib! I can't wait to try this out (react-habitat + react-habitat-redux) but I am getting a build error and wondering if I am just missing something obvious:

```
[at-loader] Checking finished with 1 errors
[at-loader] node_modules/react-habitat-redux/index.d.ts:19:15
    TS2420: Class 'Container' incorrectly implements interface 'IReduxContainer'.
  Types of property 'constructor' are incompatible.
    Type 'Function' is not assignable to type '(store: any) => IReduxContainer'.
      Type 'Function' provides no match for the signature '(store: any): IReduxContainer'
```

**Dependencies**
```
    "react": "^15.4.2",
    "react-dom": "^15.4.2",
    "react-habitat": "^0.4.1",
    "react-habitat-redux": "^1.0.1",
    "react-redux": "^5.0.2",
    "redux": "^3.6.0"
```

**Comment by theladyjaye** [comment]
It looks like `Container` just needs it's interface updated:

```
export interface IContainer {

	/**
	 * Register a new component in the container
	 * @param {string}      name        - The key that identifies this component
	 * @param {*}           comp        - The component class
	 */
	register: (name: string, comp: any) => void;

	/**
	 * Register multiple components in the container
	 * @param {object}      comps        - The components
	 */
	registerAll: (comps: {}) => void;

	/**
	 * Get a registered component for a key
	 * @param {string}      name        - The key name of the component that has been registered
	 */
	resolve: (name: string) => any;

	/**
	* The container's unique id
	*/
	id: () => string;

	/**
	 * The containers dom factory
	 */
	domFactory: () => IDomFactory;
}
```
react-habitat just got ahead of react-habitat-redux it seems?

**Comment by theladyjaye** [comment]
This might be what we need:

```
export interface IReduxContainer extends IContainer {
	/**
	 * Register a new component in the container
	 * @param {string}      name        - The key that identifies this component
	 * @param {*}           comp        - The component class
	 */
	register: (name: string, comp: any) => void;

	/**
	 * Register multiple components in the container
	 * @param {object}      comps        - The components
	 */
	registerAll: (comps: {}) => void;

	/**
	 * Get a registered component for a key
	 * @param {string}      name        - The key name of the component that has been registered
	 */
	resolve: (name: string) => any;

	/**
	* The container's unique id
	*/
	id: () => string;

	/**
	 * The containers dom factory
	 */
	domFactory: () => IDomFactory;
}

export interface ReduxStoreConstructable{
	new (store: any): IReduxContainer;
}

declare module ReduxHabitat {
	export var Container: ReduxStoreConstructable;
}
```

Note that:

```
// /**
//  * Register a component
//  */
// registerComponent: (name: string, comp: any) => void;

```

Is not present in `IContainer`

**Comment by jennasalau** [comment]
Ahh yes the interface is outdated. 

Thanks for raising the issue and investigating!

I can update it this coming weekend. If you need it sooner, please put in a PR and I can release it for you.

Thanks
Jenna

**Comment by theladyjaye** [comment]
I'll PR it for you. I just wanted to be sure that `registerComponent` was indeed supposed to be omitted. Seems to be the case.

**Comment by theladyjaye** [comment]
See: #2 

Let me know if you want me to make any adjustments, I'll be happy to.

**Comment by jennasalau** [comment]
Thanks so much @aventurella 

Yeah `registerComponent` is an old deprecated method for the simpler `register`.

This issue is now resolved in **v1.0.2** in conjunction with **react habitat v0.4.2**

Thanks for your contribution, greatly appreciated.
